components/script/dom/htmltableelement.rs

```diff
@@ -51,6 +54,31 @@ impl HTMLTableElement {
 }
 
 impl HTMLTableElementMethods for HTMLTableElement {
+    // https://html.spec.whatwg.org/multipage/#dom-table-rows
+    fn Rows(&self) -> Root<HTMLCollection> {
+        #[derive(JSTraceable, HeapSizeOf)]
+        #[allow(moved_no_move)]
+        struct TableRowFilter {
+            sections: RootedVec<JS<Node>>
+        }
+
+        impl CollectionFilter for TableRowFilter {
```

Comment:
This is doing more traversal than it needs to, but there doesn't seem to be any way of avoiding it with the HTMLCollection API.

I wonder if we could tweak HTMLCollection to take a general iterator with utility methods to create a filter iterator, cc @nox